### PR TITLE
[FIX] website_form: metadata recording

### PR DESCRIPTION
**Pull request by chienandalu**
Description of the issue/feature this PR addresses:

`website` model has this hidden field wich isn't being used at all but wich implementation becomes handy to fullfil GDPR requirement acquiring website forms metadata.

Current behavior before PR:

- There's no place where the metadata gathering can be set.
- If we set the field to True a key error raises

Desired behavior after PR is merged:

- We can set in the website settings if the metadata is gathered or not.

cc @Tecnativa

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_form/models/models.py [hunk 1]
```diff
@@ -18,6 +18,11 @@ def _website_form_last_record(self):
         return False
 
 
+class WebsiteConfigSettings(models.TransientModel):
+    _inherit = 'website.config.settings'
```
Comment (Yajo) [inline addons/website_form/models/models.py]: Runbot says this doesn't exist, can you check?

**Comment by pedrobaeza** [comment]
Check runbot status

**Comment by rafaelbn** [comment]
@chienandalu please check runbot.

As https://github.com/odoo/odoo/pull/24848 was merged then are we waiting here for the forward port @pedrobaeza ?

**Comment by Yajo** [comment]
Usually Odoo makes the forward-port by themselves, can we close this?

**Comment by chienandalu** [comment]
No, this not only fixes the bug but also adds a configuration view to set it-

**Comment by JKE-be** [comment]
Landed in master: 1f30260f853d37ad857c2a09b51e371f08194421

Thanks

**Pull request closed without merging**